src/Admin.php

```diff
@@ -277,4 +298,35 @@ public static function wp_dropdown_users_args(array $query_args) {
     return $query_args;
   }
 
+  /**
+   * @implements wp_revisions_to_keep
+   */
+  public static function wp_revisions_to_keep($num, \WP_Post $post) {
+    // Revisions for static pages (such as about or imprint pages) should be
+    // retained infinitely.
+    if ($post->post_type === 'page') {
+      $num = -1;
+    }
+    // Revisions of articles that were published a long time ago (3 months) are
+    // no longer helpful and can be cleaned up.
+    // Note: This also affects later edits unless the post's publishing date is
+    // updated.
+    elseif ($post->post_type === 'post' && time() - get_post_time('U', TRUE, $post->ID) > DAY_IN_SECONDS * static::CRON_EVENT_REVISION_CLEANUP_RETAIN_DAYS) {
+      $num = 0;
+    }
+    return $num;
+  }
+
+  /**
+   * Cron event callback to clean up obsolete revisions.
+   */
+  public static function cron_revision_cleanup($limit = 100) {
+    global $wpdb;
+
+    $post_ids = $wpdb->get_col($wpdb->prepare("SELECT ID FROM $wpdb->posts WHERE post_type = 'revision' AND post_date < %s AND post_name NOT LIKE '%autosave%' LIMIT 0,%d", date('Y-m-d', strtotime('today - ' . static::CRON_EVENT_REVISION_CLEANUP_RETAIN_DAYS . ' days')), $limit));
```

Comment:
This query does not actually limit the selected revisions to certain post types (of the actual post), so all revisions older than 3 months are deleted right now.

/cc @normanhildenbrand @ChristophKnopf